server/src/main/java/org/elasticsearch/cluster/routing/ShardRouting.java

```diff
@@ -505,7 +502,7 @@ public ShardRouting moveToStarted() {
             null,
             null,
             allocationId,
-            UNAVAILABLE_EXPECTED_SHARD_SIZE
+            expectedShardSize
```

Comment:
We talked about only keeping this for searchable snapshots, where the size is immutable.

Reply:
Done!

Reply:
I pass `Metatada` to `createStarted` in order to figure whether a new shard is searchable snapshot